src/workspace-tree/bazel_workspace_tree_provider.ts

```diff
@@ -34,12 +35,20 @@ export class BazelWorkspaceTreeProvider
 
   private disposables: vscode.Disposable[] = [];
 
+  public static fromExtensionContext(
```

Comment:
afaict, we could just add a `context: vscode.ExtensionContext` parameter to the constructor? Not sure why we would need an additional factory function

Reply:
You're right. The main intention is that in later change for making the `BazelWorkspaceTreeProvider` easier to be tested, I'm also pulling out workspace tree specific queries into a standalone `Query` interface/class, so that it would be easier to mock or fake the dependencies for the provider, like in [not yet rebased one](https://github.com/bazel-contrib/vscode-bazel/pull/422/files#diff-32978083117c650db3d83827ed159c4e5a502efcacd6676a6b5a17f3bc7dd107R48).

If this is too over engineered in this CL I can make this simpler one until it's making more sense in the later changes. WDYT?

Reply:
"easier to be tested" - makes sense. Didn't understand the underlying motivation, but now that I understand it, the code is fine as is